### PR TITLE
Add burn-peft: Parameter-Efficient Fine-Tuning (LoRA, DoRA, QLoRA, QDoRA)

### PR DESCRIPTION
**Pull request by huy209vn**
Implements  PEFT methods for Burn with full CUDA support and comprehensive testing.

Features
LoRA - Low-rank adaptation with merge/unmerge support
DoRA - Weight-decomposed LoRA with detached norm gradients
LoRA+ - Differential learning rates (16x faster convergence)
QLoRA - LoRA with 4-8 bit quantized base weights
QDoRA - DoRA with quantization
Adapter Composition - Weighted sum and concatenation
Benchmarks (CPU - NdArray backend)
Small Model (784→512→256→10):

Parameters:
  Full Model:            535,818
  LoRA Trainable:         18,640  (3.5% of full)
  
Memory (FP32):
  Gradient Savings:       96.5%  (1.97 MB saved)
  
Performance:
  Forward (merged):       0.90x  (10% faster than full!)
  Forward (unmerged):     1.10x
  Backward:               1.30x
Medium Model (784→2048→1024→10):

Parameters:
  Full Model:          3,716,106
  LoRA Trainable:         55,504  (1.5% of full)
  
Memory:
  Gradient Savings:       98.5%  (13.96 MB saved)
  QLoRA (4-bit):          86.0%  (12.19 MB saved, 14.18 MB → 1.98 MB)
Testing
✅ 22 tests pass (NdArray backend)
✅ 4 tests pass (CUDA backend - quantization verified)
✅ All clippy/fmt checks pass
✅ Working MNIST example with train/infer/bench modes
Files Changed
crates/burn-peft/ - New crate with complete PEFT implementations
examples/peft-mnist/ - Example demonstrating usage and benchmarks
Comprehensive rustdoc, README, and performance analysis included

### CODE REVIEW

**Comment by laggui** [comment]
Thanks for the contribution :slightly_smiling_face: 

/edit: just realized, I had marked this PR last week in my list since it included such a large area but I never commented.

It is better for such big features or changes to be discussed in advance, otherwise PRs are too large for us to responsibly review as-is.

In addition, there are strong signals that this was in big part AI generated (or at least, very heavily assisted).

Nothing wrong with using AI coding tools, but reviewing 4.5k additions at scale is not feasible for maintainers.

For such changes, it is better to discuss ahead of time and even ideally break it down in manageable chunks :pray:

File: crates/burn-peft/Cargo.toml
Comment (laggui) [inline crates/burn-peft/Cargo.toml:1]: For now, I suggest moving this crate to your own repository as it is an extension of burn. You can even publish it once you are entirely satisfied, and users can find it for their use case :slightly_smiling_face: 

We are planning to add such features but I am closing this for now due to the reasons mentioned in the other comment.

**Comment by huy209vn** [comment]
totally reasonable thanks.

**Comment by laggui** [comment]
> totally reasonable thanks.

Thanks for understanding!
